pyproject.toml

```diff
@@ -0,0 +1,148 @@
+[build-system]
+build-backend = "hatchling.build"
+requires = [
+  "hatchling",
+]
+
+[project]
+name = "basket-client"
+description = "A Python client for Mozilla's basket service."
+readme = "README.rst"
+keywords = [
+  "basket",
+  "mozilla",
+]
+license = {text = "BSD-3-Clause"}
+maintainers = [
+  { name = "Paul McLanahan" },
+  { name = "Rob Hudson" },
+]
+authors = [
+  { name = "Michael Kelly" },
+]
+requires-python = ">=3.8"
+classifiers = [
+  "Development Status :: 5 - Production/Stable",
+  "Environment :: Web Environment",
+  "Intended Audience :: Developers",
+  "License :: OSI Approved :: BSD License",
+  "Natural Language :: English",
+  "Programming Language :: Python",
+  "Programming Language :: Python :: 3 :: Only",
+  "Programming Language :: Python :: 3.8",
+  "Programming Language :: Python :: 3.9",
+  "Programming Language :: Python :: 3.10",
+  "Programming Language :: Python :: 3.11",
+  "Topic :: Communications",
+  "Topic :: Software Development :: Libraries",
+]
+dynamic = [
+  "version",
+]
+dependencies = [
+  "requests<3.0.0",
+]
+[project.urls]
+Documentation = "https://github.com/mozilla/basket-client#readme"
+Issues = "https://github.com/mozilla/basket-client/issues"
+Source = "https://github.com/mozilla/basket-client"
+
+[tool.hatch.version]
+path = "src/basket/__init__.py"
+
+[tool.hatch.build.targets.sdist]
+include = [
+  "docs/",
+  "src/",
+  "tests/",
+  "LICENSE",
+  "pyproject.toml",
+  "README.rst",
+]
+
+[tool.hatch.build.targets.wheel]
+packages = ["src/basket"]
+
+[tool.hatch.envs.test]
+dependencies = [
+  "coverage[toml]>=6.5",
+  "pytest",
+  "pytest-cov",
+]
+[tool.hatch.envs.test.scripts]
+test = "pytest {args:tests}"
+cov = "pytest --cov-config=pyproject.toml --cov=src/basket/ --cov-report term-missing {args:tests}"
+
+[[tool.hatch.envs.test.matrix]]
+python = ["3.8", "3.9", "3.10", "3.11"]
```

Comment:
I would add add a comment here stating that if you change this, you also need to change the Python matrix in `.github/whatever/file/it/is`.